guidelines/application_requirements.md

```diff
@@ -0,0 +1,84 @@
+# Guidelines for Application Requirements
+
+Authors: chasm@google.com, jlewi@google.com, kam.d.kasravi@intel.com
+
+## Objective
+
+Define requirements and recommendations aimed at graduating applications (e.g. controller and web apps) in Kubeflow to 1.0.
+
+## Scope
+
+This document is intended to cover applications that are deployed in the Kubernetes cluster and in particular:
+
+1. Custom resources and controllers
+1. Web applications
+
+This document is not intended to cover SDKs (e.g. fairing) or client side tools (like kfctl).
+
+## Goals
+
+The purpose of this doc is to:
+
+1. Establish criterion for graduating a Kubeflow application to a supported Kubeflow application; i.e. 1.0
+1. Set clear expectations for users regarding Kubeflow applications that are 1.0
+
+## Requirements
+
+### Configuration and deployment
+
+| Description | Category | Explanation |
+|-------------|----------|-------------|
+| Kustomize package  | Required  | <ul><li>Kubeflow has standardized on Kustomize for configuring and deploying packages </ul> |
+| Application CR  | Required  | <ul><li>Kubeflow has standardized on using the Kubernetes Application CR to provide consistent metadata and monitoring about applications <li> Application CR should be an owner of all resources so that deleting the application CR uninstalls the application </ul> |
+| ```app.kubernetes.io``` labels on every resource | Required  | <ul><li>Every resource that is part of the application should include the [labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels) recommended by Kubernetes </ul> |
```

Comment:
Give more information on labels to be added to ensure consistency? 

eg: https://github.com/kubeflow/manifests/blob/master/tf-training/tf-job-operator/overlays/application/application.yaml